### PR TITLE
Basic functionality

### PR DESCRIPTION
**Pull request by jplippi**
SongUsage.js and EightyBoi.js are disabled. They currently are unfinished/don't have a purpose.

### CODE REVIEW

File: src/parser/jobs/brd/EightyBoi.js [hunk 1]
```diff
@@ -0,0 +1,167 @@
+/**
+ * @author Yumiya
+ */
+
+import ACTIONS from 'data/ACTIONS'
+import Module from '../../core/Module'
```
Comment (NamelessFaceless) [inline src/parser/jobs/brd/EightyBoi.js]: fix your imports ty

File: src/parser/jobs/brd/SongUsage.js [hunk 2]
```diff
@@ -0,0 +1,265 @@
+/**
+ * @author Yumiya
+ */
+
+//import React, {Fragment} from 'react'
+//import {StatusLink} from 'components/ui/DbLink'
+import ACTIONS from 'data/ACTIONS'
+//import STATUSES from 'data/STATUSES'
+import Module from '../../core/Module'
```
Comment (NamelessFaceless) [inline src/parser/jobs/brd/SongUsage.js]: imports

File: src/parser/jobs/brd/StraightShot.js [hunk 3]
```diff
@@ -0,0 +1,122 @@
+/**
+ * @author Yumiya
+ */
+import React, {Fragment} from 'react'
+import Module from '../../core/Module'
```
Comment (NamelessFaceless) [inline src/parser/jobs/brd/StraightShot.js]: same

File: src/parser/jobs/brd/Util.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
+/**
+ * @author Yumiya
+ */
+import Module from '../../core/Module'
```
Comment (NamelessFaceless) [inline src/parser/jobs/brd/Util.js]: :woah:

File: src/parser/jobs/brd/Barrage.js [hunk 5]
```diff
@@ -0,0 +1,419 @@
+/**
+ * @author Yumiya
+ */
+import React, {Fragment} from 'react'
+import {Accordion, Icon, Message, List, Table} from 'semantic-ui-react'
+import Module from '../../core/Module'
```
Comment (NamelessFaceless) [inline src/parser/jobs/brd/Barrage.js]: :blobmorning:

File: src/parser/jobs/brd/CritRate.js [hunk 6]
```diff
@@ -0,0 +1,141 @@
+/**
+ * @author Yumiya
+ */
+
+import Module from '../../core/Module'
```
Comment (NamelessFaceless) [inline src/parser/jobs/brd/CritRate.js]: send help there's too many of these

File: src/parser/jobs/brd/DoTs.js [hunk 7]
```diff
@@ -0,0 +1,102 @@
+/**
+ * @author Yumiya
+ */
+import React, {Fragment} from 'react'
+import Module from '../../core/Module'
```
Comment (NamelessFaceless) [inline src/parser/jobs/brd/DoTs.js]: xd

File: src/parser/jobs/brd/Barrage.js [hunk 8]
```diff
@@ -0,0 +1,499 @@
...
+
+	// Assuming barrage was aligned with Raging Strikes. Will set to false if determined otherwise
+	aligned = true
+
+	get timestamp() {
+		return this.castEvent && this.castEvent.timestamp
+	}
+}
+
+// BarrageRule is an extention of TieredRule, with the following additions:
```
Comment (ackwell) [inline src/parser/jobs/brd/Barrage.js]: Oh boy. Here we go.

I'm not happy w/ the idea of hidden values at all. Even if it won't technically happen in this use case, it's setting precedent. What if someone gets a top-level thumbs-down but every requirement looks A-OK? The final % should be directly representing the content shown on expansion.

I'm less wary of the `reverse`, but not super on board with it.

Weights are a bit iffy, they become brittle as they always need to add up to 1. If you're absolutely _determined_ to have weightings, an implementation that isn't bounded by this should be used.

File: src/parser/jobs/brd/Barrage.js [hunk 9]
```diff
@@ -0,0 +1,499 @@
...
+								</List.Item>
+							</List.Content>
+						</List>
+					</Message>
+				</>,
+			},
+		}
+	}
+
+	// Needs to be fancy for this, since one type of severity is 'undefined'
```
Comment (ackwell) [inline src/parser/jobs/brd/Barrage.js]: It wasn't exactly designed to be used like this :blobknife:

File: src/parser/jobs/brd/CritRate.js [hunk 10]
```diff
@@ -0,0 +1,141 @@
...
+
+import Module from 'parser/core/Module'
+import STATUSES from 'data/STATUSES'
+import ACTIONS from 'data/ACTIONS'
+
+// Relevant crit buffs
+const CRIT_MODIFIERS = [
+	{
+		id: STATUSES.BATTLE_LITANY.id,
+		strenght: 0.15,
```
Comment (ackwell) [inline src/parser/jobs/brd/CritRate.js]: `strenght`

File: src/parser/jobs/brd/CritRate.js [hunk 11]
```diff
@@ -0,0 +1,141 @@
...
+		id: STATUSES.CHAIN_STRATAGEM.id,
+		strenght: 0.15,
+	},
+	{
+		id: STATUSES.CRITICAL_UP.id,
+		strenght: 0.02,
+	},
+	{
+		id: STATUSES.THE_SPEAR.id,
+		//fuck royal road
```
Comment (ackwell) [inline src/parser/jobs/brd/CritRate.js]: `core/Arcanum` tracks all RR effects and provides a strength modifier in the event.

File: src/parser/jobs/brd/CritRate.js [hunk 12]
```diff
@@ -0,0 +1,141 @@
...
+	_onDotTick(event) {
+		// Get the crit rate from DoT tick timestamps
+		if (event.tick) {
+
+			let accumulatedCritBuffs = 0
+
+			const dot = DOTS[event.ability.guid]
+
+			if (!dot) {
+				// What the fuck?
```
Comment (ackwell) [inline src/parser/jobs/brd/CritRate.js]: /shrug

Questioning event data is a sure way to insanity.

File: src/parser/jobs/brd/CritRate.js [hunk 13]
```diff
@@ -0,0 +1,141 @@
...
+			const buffList = []
+
+			for (let i = 0; i < CRIT_MODIFIERS.length; i++) {
+
+				const modifier = CRIT_MODIFIERS[i]
+
+				let status = undefined
+
+				if (modifier.id === STATUSES.CHAIN_STRATAGEM.id) {
+					status = this.enemies.getEntity(event.targetID).getStatus(modifier.id, dot.current <= event.timestamp ? dot.current : dot.last)
```
Comment (ackwell) [inline src/parser/jobs/brd/CritRate.js]: Can move that timestamp into a var so you're not repeating it.

Also a non-issue in this case probably but be wary of look-aheads.

File: src/parser/jobs/brd/DoTs.js [hunk 14]
```diff
@@ -0,0 +1,102 @@
...
+const DROP_TOLERANCE = 1000
+
+export default class DoTs extends Module {
+	static handle = 'dots'
+	static dependencies = [
+		'checklist',
+		'util',
+	]
+
+	_dotComponent = new DotComponent(STATUSES.CAUSTIC_BITE.id, STATUSES.STORMBITE.id)
```
Comment (ackwell) [inline src/parser/jobs/brd/DoTs.js]: Is this actually being used anywhere? I'm not seeing it unless I missed a side-effect.

File: src/parser/jobs/brd/EightyBoi.js [hunk 15]
```diff
@@ -0,0 +1,167 @@
+/**
```
Comment (ackwell) [inline src/parser/jobs/brd/EightyBoi.js]: This file could use with some commenting.

File: src/parser/jobs/brd/EightyBoi.js [hunk 16]
```diff
@@ -0,0 +1,167 @@
...
+	static dependencies = [
+		'invuln',
+		'downtime',
+	]
+
+	_debugEvents = []
+
+	_lastDotTickTimestamp = 0
+	_eightyBoiCasts = []
+	_songList = [
```
Comment (ackwell) [inline src/parser/jobs/brd/EightyBoi.js]: With all your classes elsewhere I'm surprised this isn't following suit. These could def be moved into a more generic structure. DRY and all that.

File: src/parser/jobs/brd/SongUptime.js [hunk 17]
```diff
@@ -41,17 +51,17 @@ export default class SongUptime extends Module {
...
-
-		//TODO: Define a threshold for song uptime
-		if (songlessPercentile > 3) {
+		if (songlessTime > songlessTolerance) {
 			this.suggestions.add(new Suggestion({
 				icon: ACTIONS.THE_WANDERERS_MINUET.icon,
-				why: `Being songless for ${songlessTime} seconds (${songlessPercentile}%)`,
-				severity: songlessPercentile > 7 ? SEVERITY.MAJOR : songlessPercentile > 5? SEVERITY.MEDIUM : SEVERITY.MINOR,
+				why: <Fragment>
+					Being songless for {songlessTime} seconds.
```
Comment (ackwell) [inline src/parser/jobs/brd/SongUptime.js]: Consider updating the `why` to be a more natural "response" - something like "You were songless for...".

File: src/parser/jobs/brd/SongUptime.js [hunk 18]
```diff
@@ -41,17 +51,17 @@ export default class SongUptime extends Module {
...
+				</Fragment>,
+				severity: songlessTime > songlessTolerance + TIER.MAJOR ? SEVERITY.MAJOR : songlessTime > songlessTolerance + TIER.MEDIUM ? SEVERITY.MEDIUM : SEVERITY.MINOR,
```
Comment (ackwell) [inline src/parser/jobs/brd/SongUptime.js]: Perfect use-case for `TieredSuggestion` :blobmorning:

File: src/parser/jobs/brd/SongUsage.js [hunk 19]
```diff
@@ -0,0 +1,260 @@
...
+
+	_lastDotTickTimestamp = 0
+	_songCastEvents = []
+	_deathEvents = []
+
+	_uptimeBlocks = []
+
+	_eightyBoiCasts = []
+
+	_songList = [
```
Comment (ackwell) [inline src/parser/jobs/brd/SongUsage.js]: What's all this doing here? Doesn't look like it's in use.

File: src/parser/jobs/brd/SongUsage.js [hunk 20]
```diff
@@ -0,0 +1,260 @@
...
+		//const fightDuration = (this.parser.fightDuration - this.downtime.getDowntime())/1000
+		this._setUptimeBlocks()
+		console.log('Fitted EightyBoi:')
+		console.log(this.eightyboi.getFormattedClosestEightyBoi())
+		console.log('Actual casts:')
+		console.log(this._formatSongList(this._songCastEvents))
+		//this._findIdealSongOrder()
+	}
+
+	_setUptimeBlocks() {
```
Comment (ackwell) [inline src/parser/jobs/brd/SongUsage.js]: Seems a bit unnecessary to invert the whole thing.

File: src/parser/jobs/brd/StraightShot.js [hunk 21]
```diff
@@ -0,0 +1,122 @@
...
+			],
+		}))
+
+		if (this._earlyStraightShotCasts.length) {
+			this.suggestions.add(new Suggestion({
+				icon: ACTIONS.STRAIGHT_SHOT.icon,
+				content: <Fragment>
+					{ACTIONS.STRAIGHT_SHOT.name} should ideally be refreshed with {STRAIGHT_SHOT_THRESHOLD / 1000} seconds or less left on the effect. More {ACTIONS.STRAIGHT_SHOT.name} casts than necessary mean less <ActionLink {...ACTIONS.HEAVY_SHOT} /> casts, which in turn mean a lower <ActionLink {...ACTIONS.REFULGENT_ARROW} /> chance.
+				</Fragment>,
+				severity: this._earlyStraightShotCasts.length <= TIER.MINOR ? SEVERITY.MINOR : this._earlyStraightShotCasts.length <= TIER.MEDIUM ? SEVERITY.MEDIUM : SEVERITY.MAJOR,
```
Comment (ackwell) [inline src/parser/jobs/brd/StraightShot.js]: `TieredSuggestion`

File: src/parser/jobs/brd/Util.js [hunk 22]
```diff
@@ -0,0 +1,74 @@
+/**
+ * @author Yumiya
+ */
+import Module from 'parser/core/Module'
+
+export default class Util extends Module {
```
Comment (ackwell) [inline src/parser/jobs/brd/Util.js]: I'm a little wary of this being a module unto itself but I get what you're doing with it. Keep in mind modules do have overhead on every event, currently.

File: src/parser/jobs/brd/Weaving.js [hunk 23]
```diff
@@ -0,0 +1,46 @@
...
+	ACTIONS.BARRAGE.id,
+	ACTIONS.EMPYREAL_ARROW.id,
+]
+
+export default class Weaving extends CoreWeaving {
+	isBadWeave(weave) {
+
+		if (weave.weaves.length === SPECIAL_WEAVE) {
+			const weaveSequence = weave.weaves.map(w => w.ability.guid)
+			if (_.isEqual(weaveSequence, BL_EA_BL) || _.isEqual(weaveSequence, BA_PP_EA) || _.isEqual(weaveSequence, BA_EA_PP) || _.isEqual(weaveSequence, PP_BA_EA)) {
```
Comment (ackwell) [inline src/parser/jobs/brd/Weaving.js]: If you move all the consts up top to just elements in an array, you could do this w/ `ALLOWED_WEAVES.some(...)`

File: src/parser/jobs/brd/SongUptime.js (hunk 18 above)
Comment (jplippi) [reply src/parser/jobs/brd/SongUptime.js]: W-when I made this, there was no TieredSuggestion :eyes:

File: src/parser/jobs/brd/DoTs.js (hunk 14 above)
Comment (jplippi) [reply src/parser/jobs/brd/DoTs.js]: Not yet, I'll comment it out. I'm experimenting with it in another branch :eyes:

File: src/parser/jobs/brd/About.js [hunk 24]
```diff
@@ -25,6 +25,6 @@ export default class About extends CoreAbout {
 		to: '4.36',
```
Comment (ackwell) [inline src/parser/jobs/brd/About.js]: Might wanna update this for 4.4 :blobmorning:

**Pull request merged**